python/openassetio/test/manager/apiComplianceSuite.py

```diff
@@ -129,3 +176,113 @@ def __assertPolicyResults(self, numSpecifications, context):
 
         self.assertValuesOfType(policies, int)
         self.assertEqual(len(policies), numSpecifications)
+
+
+class Test_isEntityReference(FixtureAugmentedTestCase):
+    """
+    Check plugin's implementation of
+    managerAPI.ManagerInterface.isEntityReference.
+
+    This will only be run if the supplied fixtures set the
+    `a_valid_entity_reference` key.
+    """
+
+    def setUp(self):
+        self._a_valid_reference = self._fixtures.get('a_valid_entity_reference')
```

Comment:
Great idea.

I've sketched something out [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/239/commits/4a144487ed52af7119ef2e049dd751be659b1e07). I updated the behaviour a little from your design - as when I put it into practice, it felt like the default of a 'require' method to skip, was perhaps backwards. See what you think.  I also added some conveniences to avoid boilerplate in common use patterns that came up whilst prototyping adoption [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/239/commits/ac77b93ef02f7d32aba3c63a677d4e573fd5f02c), [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/239/commits/5bf3109c75d97cfe9d963856f23e866e81a653cc) and [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/239/commits/82459fdbdd6965ab94703c1ff7744f8cb5657e72).

Reply:
Just occurred to me that `collectRequiredFixture` means that in practice all fixture keys must be valid variable names, which is a restriction we didn't previously have.  I guess I'm happy with that if you are?

Reply:
Cool, yeah, felt reasonable. I added some notes [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/239/commits/b3f57a2e0bbdc871b5dd3022648efc31b1752eb2).